pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFieldSpec.java

```diff
@@ -70,13 +72,19 @@ public DateTimeFieldSpec() {
    *       2) if a time column is defined in hoursSinceEpoch (format=1:HOURS:EPOCH), and the data buckets are 1 hours,
    *          the granularity will be 1:HOURS
    */
-  public DateTimeFieldSpec(String name, DataType dataType, String format, String granularity) {
+  public DateTimeFieldSpec(String name, DataType dataType, String format, String granularity,
+      @Nullable String sampleValue) {
     super(name, dataType, true);
 
     _format = format;
     _granularity = granularity;
     _formatSpec = new DateTimeFormatSpec(format);
     _granularitySpec = new DateTimeGranularitySpec(granularity);
+    _sampleValue = sampleValue;
```

Comment:
Sample value should be an `Object` (e.g. user may put long number as the sample value)